### PR TITLE
New bug fix; Also merging without_OA updates into develop

### PR DESCRIPTION
**Pull request by ayan-biswas**
New bug fix in this PR: `prefix_dict` in `netlist_config.py` repeats `mos3_cir` twice, but one of those should be `mos_cir`.
Related request: Please update the instructions in the "BAG3++ setup in a new PDK" page of the documentation to explain how to find relevant device parameters for `bag_prim.cir` (I assume it's similar to CDL?). Currently the documentation explains `bag_prim.cdl` and `bag_prim.scs` only.

This PR also merges in the previous feature addition and bug fix from `without_OA` into `develop`.

### CODE REVIEW

**Comment by ayan-biswas** [comment]
Added fix / workaround for SKILL command to replace the master library and cell names of template cells during schematic generation.

**Comment by boblinchuan** [comment]
Adding a note that merged features are from PR #43

File: run_scripts/start_bag_ICADV12d3.il
Comment (boblinchuan) [inline run_scripts/start_bag_ICADV12d3.il:1]: Trusting your fix. Do you have an idea if this is backward compatible? Or which version of Virtuoso this affects?

**Review by boblinchuan — APPROVED** [review]

File: run_scripts/start_bag_ICADV12d3.il
Comment (boblinchuan) [reply run_scripts/start_bag_ICADV12d3.il:1]: Also, do we still have a use for both `start_bag.il` and `start_bag_ICADV12d3.il`? The latter contains all the features of the former with no changes to those functions.

**Review by FeliciaG — APPROVED** [review]

**Comment by boblinchuan** [comment]
I did some digging through the SKILL references. Here's what I found:
- `schReplaceProperty` is a low-level editing function. We should probably instead use a higher-lever function like `dbSet` or `schHiReplace`
- `cellName` and `libName` are not supposed to be edited, and in fact `dbSet` cannot affect them. `schReplaceProperty` is supposed to follow `dbSet`, so the fact that we can use `schReplaceProperty` here is actually a SKILL bug.
- The better solution would be to do something like this:
`masterId = dbOpenCellViewByType(lib cell view)`
`dbSet(inst masterId "master")`
I'll work up a PR for this later.

File: run_scripts/start_bag_ICADV12d3.il
Comment (ayan-biswas) [reply run_scripts/start_bag_ICADV12d3.il:1]: The old method caused a "fprintf/sprintf" error in IC23.1. I haven't checked if the modified version is backward compatible, but it sounds like you may have found a better alternative anyway.

No, I don't think the `start_bag.il` is used right now, but we may consider having different `start_bag.il` for different IC versions if that becomes necessary at some point, for example if some old SKILL function becomes deprecated or a new non-backward-compatible SKILL function gets released.

**Pull request merged**